pkg/operator2/configsync.go

```diff
@@ -132,22 +133,30 @@ func newIDPSyncData() idpSyncData {
 
 // AddSecret initializes a sourceData object with proper data for a Secret
 // and adds it among the other secrets stored here
-// Returns the key that it stored the Secret to
-func (sd *idpSyncData) AddSecret(index int, secretName, idpType string) string {
-	dest, data := newSourceDataIDPSecret(index, secretName, idpType)
+// Returns the path for the Secret
+func (sd *idpSyncData) AddSecret(index int, secretName configv1.SecretNameReference, key string) string {
+	dest, data := newSourceDataIDPSecret(index, secretName, key)
 	sd.secrets[dest] = data
 
-	return dest
+	return data.path
+}
+
+func (sd *idpSyncData) AddSecretStringSource(index int, secretName configv1.SecretNameReference, key string) configv1.StringSource {
```

Comment:
I'd rather if we got rid of the `helpers.go` file by including the original function in `idp.go` instead of making this to be a method of this object. Having `AddSecretStringSource()` method does not make much sense because you don't have special storage for `SecretStringSource` in `idpSyncData`, there are only `Secrets` and `ConfigMaps`.